src/test/java/com/google/step/snippet/external/GeeksForGeeksClientTest.java

```diff
@@ -0,0 +1,90 @@
+package com.google.step.snippet.external;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.google.step.snippet.data.Card;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class GeeksForGeeksClientTest {
+
+  private final GeeksForGeeksClient client = new GeeksForGeeksClient("CSE_ID");
+
+  @Test
+  public void htmlCodeCard() {
```

Comment:
I think the logic for determining when to do HTML escaping should also be checked. Could you add tests where you check:
1. Query does not contain escape filter, but webpage does. In this case, the HTML should still be escaped. 
Example: `client.search("https://www.geeksforgeeks.org/how-to-change-the-height-of-br-tag/", "br tag");`
2. Query contains escape filter, but webpage does not.  In this case, the HTML should still be escaped. 
Example: `client.search("https://www.geeksforgeeks.org/how-to-display-error-without-alert-box-using-javascript/", "html display error");`
3. Neither query nor webpage contains escape filter. In this case, the HTML should not be escaped. 
Example: `client.search("https://www.geeksforgeeks.org/how-to-display-error-without-alert-box-using-javascript/", "javascript display error");`

Reply:
Sure!